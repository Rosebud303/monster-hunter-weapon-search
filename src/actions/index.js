export const setCards = (cards) => ({
  type: 'SET_CARDS',
  cards
})