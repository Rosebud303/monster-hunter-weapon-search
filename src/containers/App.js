import React, { Component } from 'react';
import { connect } from 'react-redux';

export class App extends Component {
  render() {
    return (
      <div>
        
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  cards: state.cards
});

export const mapDispatchToProps = (dispatch) => ({
  setCards: (cards) => dispatch(setCards(cards))
});

export default connect(mapStateToProps, mapDispatchToProps)(App)

