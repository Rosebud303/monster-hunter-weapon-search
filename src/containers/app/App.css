.app{
    background-image: url("https://images2.alphacoders.com/749/749574.jpg");
    background-repeat: none;
    height: 57.4em;
    background-color: #33334a;
}
