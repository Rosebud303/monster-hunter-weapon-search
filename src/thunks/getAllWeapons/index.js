import { setWeapons, isLoading } from '../../actions';
import fetchAll from '../fetches/fetchAll';

const getAllWeapons = (url) => {
  return async (dispatch) => {
    try{
      dispatch(isLoading(true))
      const weapons = await fetchAll(url)
      console.log(weapons)
      dispatch(isLoading(false))
      dispatch(setWeapons(weapons))
    } catch (error) {
      //dispatch(hasErrored(error.message))
    }
  }
}

export default getAllWeapons;