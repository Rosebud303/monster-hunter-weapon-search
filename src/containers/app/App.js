import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css'
import { Route, Switch } from 'react-router-dom'
import WeaponDetails from '../../components/weaponDetails/WeaponDetails'
import Header from '../header/Header'
import Loader from '../../components/loader/Loader'
import getAllWeapons from '../../thunks/getAllWeapons';
import WeaponContainer from '../weaponContainer/WeaponContainer'
import * as actions from '../../actions';
import FullImage from '../../components/fullImage/FullImage';
import Filter from '../../containers/filterOptions/Filter'

export class App extends Component {

  componentDidMount() {
    this.props.getAllWeapons('https://mhw-db.com/weapons')
  }

  render() {
    return (
      <div className='app'>
        <Header />
        <Filter />
        {
          !this.props.isLoading ?
          <Switch>
            <Route exact path='/' component={WeaponContainer} />
            <Route path='/weapons/:id' 
              render={({ match }) => {
                const weapon = this.props.weapons.find(weap => weap.id == parseInt(match.params.id));
                return <WeaponDetails {...weapon} />
              }} 
            />
            <Route path='/image/:name' 
              render={({ match }) => {
                const weapon = this.props.weapons.find(weap => weap.name == match.params.name);
                return <FullImage {...weapon} />
              }} 
            />
          </Switch>
          :
          <Loader />
        }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  weapons: state.weapons,
  isLoading: state.isLoading
});

export const mapDispatchToProps = (dispatch) => ({
  getAllWeapons: (url) => dispatch(getAllWeapons(url)),
  getCategoryTitles: (array) => dispatch(actions.getCategoryTitles(array))
});

export default connect(mapStateToProps, mapDispatchToProps)(App)

