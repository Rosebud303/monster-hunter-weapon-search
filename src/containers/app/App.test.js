import React from 'react';
import { shallow } from 'enzyme';
import * as actions from '../../actions';
import {App, mapDispatchToProps, mapStateToProps} from './App';

import getAllWeapons from '../../thunks/getAllWeapons';
jest.mock('../../thunks/getAllWeapons');


let mockWeps = [{
  craftable: true,
  damageType: "sever",
  id: 1,
  maxDMG: 384,
  minDMG: 80,
  name: "Buster Sword 1",
  picture: "https://assets.mhw-db.com/weapons/great-sword/83f7ab6e7e5669ec416d772049b8b054e2624c2e.c7bad811e203c9bb55626e414659a4f7.png",
  rarity: 1,
  type: "great-sword",
}, {
  craftable: true,
  damageType: "sever",
  id: 2,
  maxDMG: 384,
  minDMG: 80,
  name: "Buster Sword 2",
  picture: "https://assets.mhw-db.com/weapons/great-sword/83f7ab6e7e5669ec416d772049b8b054e2624c2e.c7bad811e203c9bb55626e414659a4f7.png",
  rarity: 1,
  type: "great-sword",
}]

describe('App', () => {
  
  describe('Component', () => {
    let wrapper;
    let mockLoad;
    let mockGetWeps;
    let mockGetCats;

    beforeEach(() => {
      mockGetWeps = jest.fn();
      mockGetCats = jest.fn();
      mockLoad = false;
      wrapper = shallow (
        <App getAllWeapons={mockGetWeps}
             isLoading={mockLoad}
             getCategoryTitles={mockGetCats}
             weapons={mockWeps}/>
      )
    });

    it('should match snap shot', () => {
      expect(wrapper).toMatchSnapshot()
    });
    
  });
  
  describe('mapStateToProps', () => {
    it('should have isLoading in state', () => {
      const mockState = {
        weapons: mockWeps,
        isLoading: false,
        error:''
      }
      const expected = {
        weapons: mockWeps,
        isLoading: false
      }
      const result = mapStateToProps(mockState);
      expect(result).toEqual(expected)
    });
    
  });

  describe('mapDispatchToProps', () => {
    it('should call getAllWeapons with the correct params', () => {
      let mockUrl = 'www.MHW.com';
      let mockDispatch = jest.fn();
      let actionsToDispatch = getAllWeapons(mockUrl);
      let mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getAllWeapons(actionsToDispatch);
      expect(mockDispatch).toHaveBeenCalledWith(actionsToDispatch);
    });
  }); 
});
