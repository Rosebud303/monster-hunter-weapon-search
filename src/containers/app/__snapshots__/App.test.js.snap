// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snap shot 1`] = `
<div
  className="app"
>
  <Header />
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WeaponContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/weapons"
    />
    <Route
      path="/weapons/:id"
      render={[Function]}
    />
    <Route
      path="/image/:name"
      render={[Function]}
    />
  </Switch>
</div>
`;
